Memorize/MemoryGame.swift

```diff
@@ -8,79 +8,81 @@
+        1. At a time only two cards should be faceUp
+        2. If the content of the card is matching then update the score by 2
+        3. else facedown all the cards and if the if the any one of the card is already seen then reduce score by 1
+        """
+        
         guard
             var chosenCard = cards.first(where: {$0.id == card.id}),
             !chosenCard.isFaceUp,
             !chosenCard.isMatched
-        else {
-            return
-        }
-        print("Chosen Card Initially \(chosenCard)")
+        else { return }
+        
         if var potentiallyMatchCard =  indexOfTheOneAndOnlyFaceUpCard {
             if chosenCard.content == potentiallyMatchCard.content {
                 chosenCard.isMatched = true
                 potentiallyMatchCard.isMatched = true
-                score += 2
-            } else {
-                // If not matched and card is already seen then score should reduce by 1
-                if chosenCard.isPreviouslySeen ||
-                   potentiallyMatchCard.isPreviouslySeen {
-                    score -= 1
-                }
+                score += 2  // If cards contain is matched then increase the score by 2
+                updateOriginalCard(chosenCard)
+                updateOriginalCard(potentiallyMatchCard)
             }
-            updateOriginalCard(chosenCard)
-            updateOriginalCard(potentiallyMatchCard)
             indexOfTheOneAndOnlyFaceUpCard = nil
         } else {
+            // If the cards are not matching then mark them as previouslySeen and face down all the cards
             for index in cards.indices {
                 if cards[index].isFaceUp {
                     cards[index].isFaceUp = false
                     cards[index].isPreviouslySeen = true
                 }
             }
             indexOfTheOneAndOnlyFaceUpCard = chosenCard
+            if chosenCard.isPreviouslySeen {
+                score -= 1
+            }
         }
         chosenCard.isFaceUp.toggle()
         updateOriginalCard(chosenCard)
     }
-
+    
     private mutating func updateOriginalCard(_ card: Card) {
-//        This function will update the cards in the original set
-        if let indexOfcardToUpdate = cards.firstIndex(where: {$0.id == card.id}) {
-            cards[indexOfcardToUpdate].isFaceUp = card.isFaceUp
-            cards[indexOfcardToUpdate].isMatched = card.isMatched
-            cards[indexOfcardToUpdate].isPreviouslySeen = card.isPreviouslySeen
-        }
-
+        """
+        This function will update the cards in the original set
+        """
+        guard let indexOfcardToUpdate = cards.firstIndex(where :{$0.id == card.id})
+        else { return }
+        cards[indexOfcardToUpdate].isFaceUp = card.isFaceUp
+        cards[indexOfcardToUpdate].isMatched = card.isMatched
+        cards[indexOfcardToUpdate].isPreviouslySeen = card.isPreviouslySeen
```

Comment:
Do you really need the index here?  Can you just find the card and deal with the card object directly?

Reply:
I tried the following code, but it is not updating the cards.
<img width="591" alt="image" src="https://user-images.githubusercontent.com/97313818/188015014-141c7196-3d0b-428e-833d-99087caf931c.png">

I think it is because of the following reason:
1. Card is an array of the struct.
2. Whenever I select a particular card from the deck of the card, a copy of that card is returned.
    And changes are being made to the copy of the card instead of the original card. 
Hence, I use the index to update the card.

Reply:
Could you please suggest to me some other approach?